transport-native-io_uring/src/main/java/io/netty/channel/uring/IOUringEventLoop.java

```diff
@@ -0,0 +1,157 @@
+                                       0,
+                                       0,
+                                       0);
+                        addNewEvent(event);
+                        ringBuffer.getIoUringSubmissionQueue().submit();
+                        break;
+                    case READ:
+                        System.out.println("Eventlloop Read Res: " + ioUringCqe.getRes());
+                        System.out.println("Eventloop Fd: " + event.getAbstractIOUringChannel().getSocket().getFd());
+                        ByteBuf byteBuf = event.getReadBuffer();
+                        int localReadAmount = ioUringCqe.getRes();
+                        if (localReadAmount > 0) {
+                            byteBuf.writerIndex(byteBuf.writerIndex() + localReadAmount);
+                        }
+
+                        final IOUringRecvByteAllocatorHandle allocHandle =
+                                (IOUringRecvByteAllocatorHandle) event.getAbstractIOUringChannel().unsafe()
+                                                                      .recvBufAllocHandle();
+                        final ChannelPipeline pipeline = event.getAbstractIOUringChannel().pipeline();
+
+                        allocHandle.lastBytesRead(localReadAmount);
+                        if (allocHandle.lastBytesRead() <= 0) {
+                            // nothing was read, release the buffer.
+                            byteBuf.release();
+                            byteBuf = null;
+                            break;
+                        }
+
+                        allocHandle.incMessagesRead(1);
+                        //readPending = false;
+                        pipeline.fireChannelRead(byteBuf);
+                        byteBuf = null;
+                        allocHandle.readComplete();
+                        pipeline.fireChannelReadComplete();
+                        event.getAbstractIOUringChannel().executeReadEvent();
+                        break;
+                    case WRITE:
+                        System.out.println("Eventloop Write Res: " + ioUringCqe.getRes());
+                        System.out.println("Eventloop Fd: " + event.getAbstractIOUringChannel().getSocket().getFd());
+                        //remove bytes
+                        int localFlushAmount = ioUringCqe.getRes();
+                        if (localFlushAmount > 0) {
+                            event.getAbstractIOUringChannel().unsafe().outboundBuffer().removeBytes(localFlushAmount);
+                        }
+                        break;
+                    }
+                } else {
+                    System.out.println("Event is null!!!! ");
+                }
+            }
+            //run tasks
+            if (hasTasks()) {
+                runAllTasks();
+            }
+            try {
+                Thread.sleep(10);
```

Comment:
@normanmaurer it's temporary, I don't quite understand how that works with the `wakeups` in the eventloop